### PR TITLE
docs(pattern): Clean up the README and type documentation

### PR DESCRIPTION
**Pull request by gibson042**
* Fix typos
* Tweak phrasing
* Add a clarifying example

### CODE REVIEW

File: packages/patterns/README.md [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
+  toString: M.callWhen(M.await(M.any())).returns(M.string()),
+});
+const asyncSerializer = makeExo('AsyncSerializer', asyncSerializerI, {
+  // M.callWhen() delays invocation of this method implementation
+  // while externally-provided input is unsettled.
+  toString(val) { return String(val); },
+});
+```
+
+See [types.js](./src/types.js) for the definitions of these new types and (at typedef `PatternMatchers`) the methods of the exported `M` namespace object.
```
Comment (erights) [inline packages/patterns/README.md:23]: Actually, the `PatternMatchers` type only lists the pattern matching methods. The overall type of the `M` object is something like `MatcherNamespace`. You should look.

**Review by erights — CHANGES_REQUESTED** [review]
Thanks!

File: packages/patterns/README.md [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
 
-The main export from the package is an `M` namespace object, for making a variety of Matchers (hence "M") but also guards, such as the InterfaceGuards used by the `@endo/exo` package to serve as the first level of defense for Exo objects --- those exposed to external messages. The InterfaceGuards use patterns for enforcing type-like restrictions on the arguments and results of these potentially malicious messages.
+The main export from the package is an `M` namespace object, for making a variety of Matchers (hence "M").
 
-See [types.js](./src/types.js) for the definitions of these new types and the methods of the exported `M` namespace object.
+`M` can also make _Guards_ that use Patterns to characterize dynamic behavior such as method argument/response signatures and promise awaiting. The `@endo/exo` package uses InterfaceGuards (each of which maps a collection of method names to their corresponding method guards) as the first level of defense for Exo objects against malformed input. For example:
+```js
+const asyncSerializerI = M.interface('AsyncSerializer', {
+  toString: M.callWhen(M.await(M.any())).returns(M.string()),
```
[the remark this answers is not in the document]
Comment (gibson042) [reply packages/patterns/README.md]: Ah right, collision with the `Object.prototype` method. Fixed.

File: packages/patterns/README.md (hunk 1 above)
Comment (gibson042) [reply packages/patterns/README.md]: I did look, but forgot the separation by the time I was making edits here. :upside_down_face:

Fixed.

**Review by erights — APPROVED** [review]
LGTM

**Pull request merged**